src/engine/guidance/post_processing.cpp

```diff
@@ -1037,6 +1039,9 @@ std::vector<RouteStep> collapseTurns(std::vector<RouteStep> steps)
     // a series of turns is only possible to collapse if its only name changes and suppressed turns.
     const auto canCollapseAll = [&steps](std::size_t index, const std::size_t end_index) {
         BOOST_ASSERT(end_index <= steps.size());
+        if (!compatible(steps[index], steps[index + 1]))
+            return false;
+        ++index;
```

Comment:
Wait wait wait - this not only adds a test but changes behavior. Why the + 1 here?